infrastructure/lib/constructs/database.ts

```diff
@@ -0,0 +1,260 @@
+        retention: cdk.Duration.days(7),
+      },
+      preferredMaintenanceWindow: "Sun:07:00-Sun:08:00",
+      parameterGroup: parameters,
+      copyTagsToSnapshot: true,
+      iamAuthentication: true,
+      storageEncrypted: true,
+      cloudwatchLogsExports: ["postgresql"],
+    });
+    this.cluster = cluster;
+
+    // Automatic secrets rotation is not yet supported via the CDK in GovCloud even
+    // though the underlying resources do exist. There is a PR open against the CDK
+    // upstream in order to add this functionality:
+    //    https://github.com/aws/aws-cdk/pull/17673
+    // cluster.addRotationSingleUser({
+    //   // Default value is 30 days, so this makes the secret much shorter-lived.
+    //   // The primary means for access will be via IAM; however, we will use this
+    //   // for initial bootstrapping and rare maintenance.
+    //   automaticallyAfter: cdk.Duration.days(7),
+    // });
+    // This secret will be present since we're using the default configuration; however,
+    // it will require additional workarounds (or the featuring be added) to rotate
+    // automatically.
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    this.adminSecret = cluster.secret!;
+
+    const bootstrapper = new DatabaseBootstrapper(this, "DatabaseBootstrapper", {
+      ...props,
+      secretName: this.adminSecret.secretName,
+      cluster: cluster,
+    });
+    this.adminSecret.grantRead(bootstrapper.backingLambda);
+    cluster.connections.allowDefaultPortFrom(bootstrapper.backingLambda);
+
+    // Ensure the DB is not created until all instances in the cluster are ready
+    const instances = cluster.node.children.filter((child) => child instanceof rds.CfnDBInstance);
+    const instancesReady = new cdk.ConcreteDependable();
+    instances.forEach((dbInstance) => instancesReady.add(dbInstance));
+    bootstrapper.bootstrapResource.node.addDependency(instancesReady);
+
+    // Provide a resource to determine whether the database has been created.
+    this.databaseReady = new cdk.ConcreteDependable();
+    this.databaseReady.add(bootstrapper.bootstrapResource);
+
+    this.clusterResourceId = this.clusterIdGetterCustomResource();
+  }
+
+  public get clusterArn(): string {
+    return cdk.Stack.of(this).formatArn({
+      arnFormat: cdk.ArnFormat.COLON_RESOURCE_NAME,
+      service: "rds",
+      resource: "cluster",
+      resourceName: this.cluster.clusterIdentifier,
+    });
+  }
+
+  private clusterUserArn(user: string): string {
+    return cdk.Stack.of(this).formatArn({
+      arnFormat: cdk.ArnFormat.COLON_RESOURCE_NAME,
+      service: "rds-db",
+      resource: "dbuser",
+      resourceName: `${this.clusterResourceId}/${user}`,
+    });
+  }
+
+  /**
+   * Grant network and IAM permissions necessary to read from the database.
+   *
+   * @param resource The resource to grant access to
+   */
+  public grantRead(resource: ec2.IConnectable & iam.IGrantable): void {
```

Comment:
Since these grants are all the same for each account, it might make sense to cut down on boilerplate by having them share a utility function.

Reply:
Yeah. I'll create a `private grantUser(resource, userName)` function